flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -234,6 +265,11 @@ public void abortCheckpointOnBarrier(
                 () -> operatorChain.broadcastEvent(new CancelCheckpointMarker(checkpointId)));
```

Comment:
>  Should we move these 4 lines into runThrowing?
That's a good question. This method `abortCheckpointOnBarrier` is called only from the `CheckpointBarrierHandler`, which is only used with network input tasks, for which `actionExecutor` is always a no-op (it's being used only for legacy sources in `SourceStreamTask`, which doesn't have network input). So I think it doesn't matter really matter, but conceptually yes, it would be technically more correct to wrap all of those lines with `actionExecutor`. 

> And should we call cancelAlignmentTimer(); in close method?

Probably yes, to wake up anyone that might be waiting on the timer.